specs/agents/metadata.md

```diff
@@ -96,8 +96,7 @@ On Linux, the container ID and some of the Kubernetes metadata can be extracted
 
     If there is a match to either expression, the capturing group contains the pod ID. We then unescape underscores 
     (`_`) to hyphens (`-`) in the pod UID.
-    If we match a pod UID then we record the hostname as the pod name since, by default, Kubernetes will set the 
-    hostname to the pod name. Finally, we record the basename as the container ID without any further checks.
+    Finally, we record the basename as the container ID without any further checks.
```

Comment:
Doesn't this change the intent?  Currently, at least in the Node.js APM agent, we only set a value for the Kubernetes pod name (for `metadata.system.kubernetes.pod.name`) in any of the following conditions:
- If the `KUBERNETES_POD_NAME` envvar is set, then use that value.
- If the `KUBERNETES_POD_UID` envvar is set, then use the hostname.
- If a k8s pod UID is parsed from a `/proc/self/cgroup` line with the steps above, then use the hostname.

So far, the changes in the PR make it sound like we would always set `metadata.system.kubernetes.pod.name = hostname`.  Did I misread?

Reply:
This part of the spec says that we set the pod name to the `KUBERNETES_POD_NAME` env variable, if it's set. Is there something that's not clear about the sentence?

> Sometimes the `KUBERNETES_POD_NAME` is set using the [Downward API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/),
> so we set the pod name to its value if it exists. Otherwise, we set the pod name to the hostname, because that's the default in Kubernetes.

Reply:
Assume that no `KUBERNETES_` envvars are set. My read of the changes in this PR is that the spec would suggest that agents *always* set `metadata.system.kubernetes.pod.name` to the hostname.  However, my read of the current spec is that `metadata.system.kubernetes.pod.name` should only be set "If we match a pod UID" in a "/proc/self/cgroup" line.

Reply:
Thanks for clarifying; I think I've restored the original logic, can you check the latest change?
Also, where in the spec did you get the condition `If the KUBERNETES_POD_UID envvar is set, then use the hostname.`?